src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -1446,6 +1446,168 @@ public void testFailedToResumeTransaction() throws Exception {
+                        assert (expected3.equals(rs.getObject(3)));
+                        assert (expected4.equals(rs.getObject(4)));
+                    }
+                }
+            }
+        }
+
+        /**
+         * Tests result of math operation in prepared statement using addition
+         * 
+         * @throws SQLException
+         *         when an error occurs
+         */
+        @Test
+        public void testMathBigDecimalAddition() throws SQLException {
+            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
+                stmt.executeUpdate("create table " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (test_column decimal(10,5))");
+                stmt.executeUpdate("insert into " + AbstractSQLGenerator.escapeIdentifier(tableName) + " values (99999.12345)");
+                try (PreparedStatement pstmt = con.prepareStatement("select (test_column + ?), " 
+                    + "(test_column + ?), (test_column + ?), (test_column + ?) from " + AbstractSQLGenerator.escapeIdentifier(tableName))) {
+                    BigDecimal value1 = new BigDecimal("1.5");
+                    pstmt.setObject(1, value1);
+                    BigDecimal value2 = new BigDecimal("0");
+                    pstmt.setObject(2, value2);
+                    BigDecimal value3 = new BigDecimal("99999.12345");
+                    pstmt.setObject(3, value3);
+                    BigDecimal value4 = new BigDecimal("99999.2");
+                    pstmt.setObject(4, value4);
+
+                    BigDecimal base = new BigDecimal("99999.12345");
+                    BigDecimal expected1 = base.add(value1);
+                    BigDecimal expected2 = base.add(value2);
+                    BigDecimal expected3 = base.add(value3);
+                    BigDecimal expected4 = base.add(value4);
+
+                    try (ResultSet rs = pstmt.executeQuery()) {
+                        rs.next();
+                        assert (expected1.equals(rs.getObject(1)));
+                        assert (expected2.equals(rs.getObject(2)));
+                        assert (expected3.equals(rs.getObject(3)));
+                        assert (expected4.equals(rs.getObject(4)));
+                    }
+                }
+            }
+        }
+
+        /**
+         * Tests result of math operation in prepared statement using multiplication
+         * 
+         * @throws SQLException
+         *         when an error occurs
+         */
+        @Test
+        public void testMathBigDecimalMultiplication() throws SQLException {
+            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
+                stmt.executeUpdate("create table " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (test_column decimal(10,5))");
+                stmt.executeUpdate("insert into " + AbstractSQLGenerator.escapeIdentifier(tableName) + " values (99999.12345)");
```

Comment:
can we not do this AbstractSQLGenerator.escapeIdentifier multiple times all over the place and just do that once?